libs/wingsdk/src/target-tf-aws/dynamodb-table.ts

```diff
@@ -0,0 +1,161 @@
+import { Construct } from "constructs";
+import { Function } from "./function";
+import { DynamodbTable as AwsDynamodbTable } from "../.gen/providers/aws/dynamodb-table";
+import * as core from "../core";
+import * as ex from "../ex";
+import { NameOptions, ResourceNames } from "../shared/resource-names";
+import { IInflightHost } from "../std";
+
+/**
+ * Table names must be between 3 and 255 characters.
+ * You can use alphanumeric characters, dot (.), dash (-), and underscores (_).
+ */
+const NAME_OPTS: NameOptions = {
+  maxLen: 255,
+  disallowedRegex: /[^a-zA-Z0-9\_\.\-]+/g,
+};
+
+/**
+ * AWS implementation of `ex.DynamodbTable`.
+ *
+ * @inflight `@winglang/sdk.ex.IDynamodbTableClient`
+ */
+export class DynamodbTable extends ex.DynamodbTable {
+  private readonly table: AwsDynamodbTable;
+
+  constructor(scope: Construct, id: string, props: ex.DynamodbTableProps) {
+    super(scope, id, props);
+
+    const hashKey = Object.entries(props.keySchema).find(
+      ([_, v]) => v === "HASH"
+    );
+    if (!hashKey) {
+      throw Error("missing HASH key in `keySchema`.");
+    }
+
+    const rangeKey = Object.entries(props.keySchema).find(
+      ([_, v]) => v === "RANGE"
+    );
```

Comment:
Should this error handling logic be shared by both targets? Maybe put this in a separate `validateKeySchema` function (checking that all values are valid, there's no more than one HASH or RANGE, etc.)